stable/external-dns/templates/service.yaml

```diff
@@ -27,5 +30,12 @@ spec:
     {{- if and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort)) }}
     nodePort: {{ .Values.service.nodePort  }}
     {{- end }}
+  - name: metrics
```

Comment:
My only concern here is that this change deviates from upstream, but we can merge this first and then see if we can contribute something back upstream to make it work for us (since we are switching to our fork for the release anyways)

Reply:
hmm does this work? I tried manually adding this new `metrics` endpoint to the running service but I'm getting `Service "external-dns-kubeaddons" is invalid: spec.ports[1]: Duplicate value: core.ServicePort{Name:"", Protocol:"TCP", AppProtocol:(*string)(nil), Port:7979, TargetPort:intstr.IntOrString{Type:0, IntVal:0, StrVal:""}, NodePort:0}`, I don't think we can specify two endpoints for the same port

Reply:
Why are we defining duplicates here?

Reply:
I think the intention was to reuse the existing servicemonitor endpoint targeting port named `metrics` in prometheus here https://github.com/mesosphere/kubernetes-base-addons/blob/master/addons/prometheus/prometheus.yaml#L93-L94 but since we can't define dupe ports in the service we can either add `port: http` to the list of prom endpoints, or `targetPort: 7979` which is likely unique to external-dns service. I tested adding `http`, it works but also pulls in thanos query endpoints so to resolve that we should then remove the thanos query target https://github.com/mesosphere/kubernetes-base-addons/blob/master/addons/prometheus/prometheus.yaml#L97-L102

Reply:
I'm going to revert this change

Reply:
We can decide how to pull this into prom in the KBA PR